src/components/elements/controlledDisplayFiltersGroup/index.tsx

```diff
@@ -23,28 +23,71 @@ export function ControlledDisplayFiltersGroup({
   label
 }: ControlledDisplayFiltersGroupProps) {
   const [showAll, setShowAll] = useState<boolean>(false);
+  const [displayedFilters, setDisplayedFilters] = useState<Set<string>>(() => new Set<string>());
+
+  const availableFiltersNames= new Set(filters.map((filter)=>filter.label));
 
   const toggleAll = () => {
+    if(!showAll){
+      setDisplayedFilters(availableFiltersNames);
+    }
+    else{
+      setDisplayedFilters(new Set());
+    }
     setShowAll(!showAll);
     toggleShowAll(!showAll);
   };
 
-  const updateFilter = (name: string, show: boolean) => {
-    onFilterChange(name, show);
+  function setsAreEqual(setA:Set<string>, setB:Set<string>) {
+    if (setA.size !== setB.size) {
+      return false;
+    }
+    for (const item of setA) {
+      if (!setB.has(item)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  useEffect(() => {
+    // When metadataToDisplay changes, check if it's equal to availableFiltersNames
+    if (setsAreEqual(displayedFilters, availableFiltersNames)) {
+      setShowAll(true);
+    } else {
+      setShowAll(false);
+    }
+  }, [displayedFilters, availableFiltersNames]);
+
+  const updateFilter = (item: DisplayFilterPartialProps, show: boolean) => {
+    if(item.onChange){
+      item.onChange(show)
+    }
+    onFilterChange(item.label, show);
   };
 
   return (
     <Box sx={{ backgroundColor: 'rgba(248, 250, 252, 1)' }}>
       <Box>
-        <LabeledSwitch label={label} onChange={toggleAll} />
+        <LabeledSwitch label={label} onChange={toggleAll} checked={showAll}/>
       </Box>
       {filters.map((item) => {
         return (
           <>
             <DisplayFilter
-              value={showAll}
+              value={displayedFilters.has(item.label)}
               label={item.label}
-              onChange={(show) => updateFilter(item.label, show)}
+              onChange={(show) => {
+                const updatedFilters = new Set(displayedFilters);
+                if (updatedFilters.has(item.label)) {
+                  updatedFilters.delete(item.label);
+                } else {
+                  updatedFilters.add(item.label);
+                }
+                setDisplayedFilters(updatedFilters);
+
+                updateFilter(item, show)
```

Comment:
This call stack is very strange:

1. DisplayFilter.onChange gets called
3. Calls updateFilter
4. updateFilter calls item.onChange (callback to higher level component)
5. updateFilter calls onFilterChange (callback to higher level component)

Questions:
1. Why does updateFilter need to exist? Can it be inline in DisplayFilter.onChange?
2. Why does are there 2 separate callbacks to update the state in the higher level component? That looks like a potential bug where you split up state to multiple locations, and they could easily get out of sync with each other (if not right now then when the code changes later)

Reply:
I agree, I removed one of the callbacks and kept only the item.onChange